backend/tests/v5/e2e/routes/teamspaces/projects/containers/containers.test.js

```diff
@@ -184,6 +185,91 @@ const testGetContainerStats = () => {
 	});
 };
 
+const testAddContainer = () => {
+	const route = `/v5/teamspaces/${teamspace}/projects/${project.id}/containers`;
+	describe('Add container', () => {
+		test('should fail without a valid session', async () => {
+			const res = await agent.post(route).expect(templates.notLoggedIn.status);
+			expect(res.body.code).toEqual(templates.notLoggedIn.code);
+		});
+
+		test('should fail if the user is not a member of the teamspace', async () => {
+			const res = await agent.post(`${route}?key=${nobody.apiKey}`).expect(templates.teamspaceNotFound.status);
+			expect(res.body.code).toEqual(templates.teamspaceNotFound.code);
+		});
+
+		test('should fail if container name already exists', async () => {
```

Comment:
you don't need all these. As your unit test on the middleware should cater for them.

Just have 1 test for invalid input to avoid cluttering the e2e.